nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/ui/web/auth/NuxeoAuthenticationFilter.java

```diff
@@ -994,7 +1014,26 @@ private boolean handleLogin(HttpServletRequest httpRequest, HttpServletResponse
                 if (descriptor.getNeedStartingURLSaving()) {
                     saveRequestedURLBeforeRedirect(httpRequest, httpResponse);
                 }
-                return Boolean.TRUE.equals(plugin.handleLoginPrompt(httpRequest, httpResponse, baseURL));
+
+                HttpServletResponse response = new HttpServletResponseWrapper(httpResponse) {
+                    @Override
+                    public void sendRedirect(String location) throws IOException {
+                        HttpServletResponse response = (HttpServletResponse) getResponse();
+                        StringBuilder sb = new StringBuilder();
+                        sb.append("<script type=\"text/javascript\">\n");
+                        sb.append("document.cookie = '" + NXAuthConstants.START_PAGE_FRAGMENT_KEY
+                                + "=' + window.location.hash.substring(1) + '; path=/';\n");
+                        sb.append("window.location = '" + location + "';\n");
```

Comment:
`window.location.hash.substring(1)` and `location` must be escaped to avoid XSS.

Reply:
Done